docs/program/sdks/use-extra-params.md

```diff
@@ -0,0 +1,117 @@
+---
+title: "Using Extra Params with Viam's SDKs"
+linkTitle: "Extra Params"
+weight: 40
+type: "docs"
+description: "Using extra parameters with Viam's SDKs."
+icon: "/services/img/icons/sdk.svg"
+tags: ["sdk", "extra", "extend"]
+---
+
+Many component API methods have the option to pass in `extra` parameters.
+These are typed as `Optional[Dict[str, Any]]` in the Python SDK and `map[string]interface{}` in the Go SDK.
+
+Here's how to [pass correctly typed objects to](#define) and [utilize](#utilize) these `extra` parameters.
+
+## Define
```

Comment:
If we can come  up with an example from a use case that would be better I think

Reply:
@npentrel Hmmm, I think I can add the code from the ticket: https://viam.atlassian.net/jira/software/c/projects/DOCS/boards/31?modal=detail&selectedIssue=DOCS-214&quickFilter=40 to the "Utilize" section and adapt define-- will try that out

Reply:
@npentrel Updated and reordered, LMK what you think!

Reply:
@npentrel Going to page eric for review on this and based on his feedback, I also think it'd be a good idea to have it in /extend instead, and link to this from within the modular resources documentation?

Reply:
@npentrel Eric recommended @randhid as a good resource for this!

Reply:
Not sure if I was pinged for a better use case or to verify that this works?

I usually do 

```go
out, ok:= extra["fieldname"].type //type is bool, string, something
if !ok {
return error
}
if out == something {
   doAfunction()
}
```

Which is equivalent to your go code example.

Reply:
Let me know how I can be more helpful

Reply:
@randhid No worries, just wanted you to verify the page's accuracy!